### PR TITLE
Use hardware limits where appropriate for glsl-to-spirv generation

### PR DESCRIPTION
**Pull request by mark-lunarg**
The test framework commonly converts glsl to spirv when creating  `VkShaderObj `objects. Glslang validates GL limits (see the list below) using internal, default limits.  The desktop interface to glslang passes in another set of default values.  However, neither the default glslang values nor the version created by the test framework use the actual hardware values for common items.  This PR passes in the current `PhysicalDeviceLimits `values so that glslang's validation will not generate inappropriate limits violation errors.

Also restored the `ExceedMaxGeometryInputOutputComponents `test for Android, since it will no longer generate invalid limits violations.

**Vulkan PDev limits which overlap with glslang's default OpenGL limits**:

- maxClipDistances;
- maxComputeWorkGroupCountX;
- maxComputeWorkGroupCountY;
- maxComputeWorkGroupCountZ;
- maxComputeWorkGroupSizeX;
- maxComputeWorkGroupSizeY;
- maxComputeWorkGroupSizeZ;
- maxCullDistances;
- maxGeometryInputComponents;
- maxGeometryOutputComponents;
- maxGeometryOutputVertices;
- maxGeometryTotalOutputComponents;
- maxVertexOutputComponents;
- maxViewports;

### CODE REVIEW

File: tests/vktestframeworkandroid.cpp [hunk 1]
```diff
@@ -111,6 +111,23 @@ bool VkTestFramework::GLSLtoSPV(const VkShaderStageFlagBits shader_type, const c
...
+    options.SetLimit(shaderc_limit_max_compute_work_group_count_z, device_limits->maxComputeWorkGroupCount[2]);
+    options.SetLimit(shaderc_limit_max_compute_work_group_size_x, device_limits->maxComputeWorkGroupSize[0]);
+    options.SetLimit(shaderc_limit_max_compute_work_group_size_y, device_limits->maxComputeWorkGroupSize[1]);
+    options.SetLimit(shaderc_limit_max_compute_work_group_size_z, device_limits->maxComputeWorkGroupSize[2]);
+    options.SetLimit(shaderc_limit_max_cull_distances, device_limits->maxCullDistances);
+    options.SetLimit(shaderc_limit_max_geometry_input_components, device_limits->maxGeometryInputComponents);
+    options.SetLimit(shaderc_limit_max_geometry_output_components, device_limits->maxGeometryOutputComponents);
+    options.SetLimit(shaderc_limit_max_geometry_output_vertices, device_limits->maxGeometryOutputVertices);
+    options.SetLimit(shaderc_limit_max_geometry_total_output_components, device_limits->maxGeometryTotalOutputComponents);
+    options.SetLimit(shaderc_limit_max_geometry_total_output_components, 4400);
```
Comment (jeremy-lunarg) [inline tests/vktestframeworkandroid.cpp]: What's this line about?

File: tests/vktestframework.cpp [hunk 2]
```diff
@@ -338,14 +338,16 @@ bool VkTestFramework::SetConfigFile(const std::string &name) {
 //
 // Parse either a .conf file provided by the user or the default string above.
 //
-void VkTestFramework::ProcessConfigFile() {
+void VkTestFramework::ProcessConfigFile(const VkPhysicalDeviceLimits *const device_limits) {
     char **configStrings = 0;
     char *config = 0;
+    bool config_file_specified = false;
```
Comment (jeremy-lunarg) [inline tests/vktestframework.cpp]: Do you need the bool? Can you check `if(!config)` instead?

File: tests/vktestframework.cpp [hunk 3]
```diff
@@ -460,9 +462,10 @@ void VkTestFramework::ProcessConfigFile() {
         else if (strcmp(token, "MaxGeometryTextureImageUnits") == 0)
             Resources.maxGeometryTextureImageUnits = value;
```
Comment (jeremy-lunarg) [inline tests/vktestframework.cpp]: Why don't we care about the other `max` limits?

File: tests/vktestframework.cpp [hunk 4]
```diff
@@ -338,14 +338,16 @@ bool VkTestFramework::SetConfigFile(const std::string &name) {
 //
 // Parse either a .conf file provided by the user or the default string above.
 //
-void VkTestFramework::ProcessConfigFile() {
+void VkTestFramework::ProcessConfigFile(const VkPhysicalDeviceLimits *const device_limits) {
```
Comment (jeremy-lunarg) [inline tests/vktestframework.cpp]: west-const... *blech*

File: tests/vktestframeworkandroid.cpp (hunk 1 above)
Comment (mark-lunarg) [reply tests/vktestframeworkandroid.cpp]: Leftover testing crumbs, thanks for finding it.

File: tests/vktestframework.cpp (hunk 2 above)
Comment (mark-lunarg) [reply tests/vktestframework.cpp]: Didn't want to override even if a bad config file was specified,

File: tests/vktestframework.cpp (hunk 3 above)
Comment (mark-lunarg) [reply tests/vktestframework.cpp]: They are not Vulkan limits.

File: tests/vktestframework.cpp (hunk 4 above)
Comment (mark-lunarg) [reply tests/vktestframework.cpp]: Will change to east-const just for you Jeremy.

**Pull request merged**